### DISCUSSION TITLE
multilanguage support

### DISCUSSION THREAD

**Discussion by Arman-Espiar** [jeffdapaz/VisualChatGPTStudio#8] [Ideas]
Hi, please multilanguage to this extension for Multilanguage performance. for example, when you select a code and click on Add comment, English comments added by default. because in **_"AddComments"_** file write "_Rewrite the code with comments..._" and chatGPT return English comments and if write with other languages ex: Farsi **_"...کد را با کامنت فارسی بازنویسی کنید"_** Or **_"Rewrite the code with Farsi comments..."_** chatGPT return comments with specify languages (in ex is Farsi).
so, i add resources file English and Farsi and change hard code to static class and change Command files. But, I haven't made a VSIX project yet, that's why I'm putting the files here for you, if you like, add them to the project.

There is no need to change all the sections, only the sections related to the explanation are more understandable, such as comments, explanations, etc

[commands and languages](https://ufile.io/l3fp6i3n)

**Comment by jeffdapaz** [comment]
Hi Arman, first of all thank you for your collaboration!

I've had a look at those files you posted and to me your implementation looks fine.

But by the way, I already thought about making the extension multilingual, and the easiest way I found is to be able to customize each  commands through the options section. This way anyone can rewrite the commands with any language, not only that, can customize the commands for any specific reason or purpose.

With this approach the project does not need to have resource files for each language or publish a new version for any new language. This approach can be more "open" for anyone.

What do you think?

**Reply by Arman-Espiar** [reply]
Good, but does this approach require editing the VSIX file? If yes, what about those who don't know how to work with VSIX projects like me? If the answer is no, how should I implement other languages?

**Reply by jeffdapaz** [reply]
No, won't be necessary edit the VSIX file.

What I was trying to say is to add the possibility to edit the commands through the options, like this:

![image](https://user-images.githubusercontent.com/63928228/225157830-bee6be7b-a57f-48a9-b310-a7fb215af16b.png)

For example, now you can come here and translate the commands, and others can translate for themselves in other languages.

But don't worry, I already implemented it and published the code in the repository. And I will publish this today on the marketplace.

**Reply by Arman-Espiar** [reply]
Wow nice, Thank yout
